clients/java/src/main/java/io/camunda/zeebe/client/impl/oauth/OAuthCredentialsProvider.java

```diff
@@ -170,4 +191,41 @@ private ZeebeClientCredentials fetchCredentials() throws IOException {
       return fetchedCredentials;
     }
   }
+
+  private void maybeConfigureCustomSSLContext(final HttpURLConnection connection) {
+    if (connection instanceof HttpsURLConnection) {
+      final HttpsURLConnection httpsConnection = (HttpsURLConnection) connection;
+      httpsConnection.setSSLSocketFactory(createSSLContext());
+    }
+  }
+
+  private SSLSocketFactory createSSLContext() {
```

Comment:
🔧 Just a little refactor

```suggestion
  private SSLSocketFactory createSSLContext() {
    if (keystorePath == null && truststorePath == null) {
      return SSLContexts.createSystemDefault().getSocketFactory();
    }
    final SSLContextBuilder builder = SSLContexts.custom();
    try {
      if (keystorePath != null) {
        builder.loadKeyMaterial(
            keystorePath,
            keystorePassword == null ? null : keystorePassword.toCharArray(),
            keystoreKeyPassword == null ? new char[0] : keystoreKeyPassword.toCharArray());
      }

      if (truststorePath != null) {
        builder.loadTrustMaterial(
            truststorePath, truststorePassword == null ? null : truststorePassword.toCharArray());
      }

      return builder.build().getSocketFactory();

    } catch (final NoSuchAlgorithmException 
                   | KeyManagementException 
                   | KeyStoreException 
                   | UnrecoverableKeyException 
                   | CertificateException 
                   | IOException e) {
      throw new RuntimeException("Failed to create SSL context", e);
    }
  }
```

Reply:
Applied your suggestion. Thank you!